From 61ff1491e80c7199bc79aa442cdcfb55986e774b Mon Sep 17 00:00:00 2001
From: Mystro256 <alexjnewt@hotmail.com>
Date: Tue, 18 Jan 2022 12:50:05 -0500
Subject: [PATCH] Use FHS compliant install

FHS does not allow installing to "amdgcn", but "LIBDIR/amdgcn" appears to be
acceptable to the FHS and distros. This logic uses GNUInstallDirs to detect the
LIBDIR instead of hardcoding it to "lib".

E.g. Fedora/Redhat use lib64 instead of lib.
---
 cmake/OCL.cmake      | 3 ++-
 cmake/Packages.cmake | 5 +++--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/cmake/OCL.cmake b/cmake/OCL.cmake
index 4e324e8..5bde654 100644
--- a/cmake/OCL.cmake
+++ b/cmake/OCL.cmake
@@ -10,6 +10,7 @@
 # configure_file and we are included from multiple places. To get around this
 # we `file(WRITE)` a file with an @variable reference and `configure_file` it.
 cmake_policy(SET CMP0053 OLD)
+include(GNUInstallDirs)
 
 if (WIN32)
   set(EXE_SUFFIX ".exe")
@@ -33,7 +34,7 @@ set (BC_EXT .bc)
 set (LIB_SUFFIX ".lib${BC_EXT}")
 set (STRIP_SUFFIX ".strip${BC_EXT}")
 set (FINAL_SUFFIX "${BC_EXT}")
-set (INSTALL_ROOT_SUFFIX "amdgcn/bitcode")
+set (INSTALL_ROOT_SUFFIX "${CMAKE_INSTALL_LIBDIR}/amdgcn/bitcode")
 
 # Set `inc_options` to contain Clang command-line for include directories for
 # current source directory.
diff --git a/cmake/Packages.cmake b/cmake/Packages.cmake
index 715ed95..e948e96 100644
--- a/cmake/Packages.cmake
+++ b/cmake/Packages.cmake
@@ -1,4 +1,5 @@
-set(PACKAGE_PREFIX lib/cmake/AMDDeviceLibs)
+include(GNUInstallDirs)
+set(PACKAGE_PREFIX ${CMAKE_INSTALL_LIBDIR}/cmake/AMDDeviceLibs)
 
 # Generate the build-tree package.
 # We know the absolute path to the build tree, so we leave
@@ -16,7 +17,7 @@ configure_file(AMDDeviceLibsConfig.cmake.in
   @ONLY)
 
 
-set(install_path_suffix "amdgcn/bitcode")
+set(install_path_suffix "${CMAKE_INSTALL_LIBDIR}/amdgcn/bitcode")
 
 # Generate the install-tree package.
 # We do not know the absolute path to the intall tree until we are installed,
-- 
2.31.1

